packages/blade/src/components/ActionList/ActionListFooter.tsx

```diff
@@ -0,0 +1,106 @@
+
+type ActionListFooterProps = {
+  title?: string;
+  leading?: React.ReactNode;
+  trailing?: React.ReactNode;
+};
+
+const StyledActionListFooter = styled(Box)((props) => {
+  return {
+    display: 'flex',
+    flexDirection: 'row',
+    alignItems: 'center',
+    padding: `${makeSize(props.theme.spacing[3])} ${makeSize(props.theme.spacing[5])}`,
+    backgroundColor: props.theme.colors.brand.gray.a50.lowContrast,
+    margin: `${makeSize(props.theme.spacing[3])} -${makeSize(props.theme.spacing[3])} -${makeSize(
+      props.theme.spacing[3],
+    )}`,
+  };
+});
+
+const ActionListFooter = (props: ActionListFooterProps): JSX.Element => {
+  const footerRef = React.useRef<HTMLDivElement | null>(null);
+  const {
+    setShouldIgnoreBlur,
+    setHasFooterAction,
+    onSelectKeydown,
+    activeIndex,
+    setIsOpen,
+  } = useDropdown();
+
+  React.useEffect(() => {
+    if (!isReactNative() && footerRef.current?.querySelector('button, a')) {
+      setHasFooterAction(true);
+    }
+  }, [setHasFooterAction, props.trailing]);
+
+  const isOnlyActionButton = !props.title && !props.leading && props.trailing;
+
+  return (
+    <StyledActionListFooter
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      ref={footerRef as any}
+      // @ts-ignore: Ignoring because the TS fails for React Native and works for web
+      onMouseDown={() => {
+        setShouldIgnoreBlur(true);
+      }}
+      // @ts-ignore: Ignoring because the TS fails for React Native and works for web
+      onKeyDown={(e) => {
+        const nativeEvent = e.nativeEvent;
+        // We ignore the selection keydowns on footer to let users click on items
+        if ((nativeEvent.key === ' ' || nativeEvent.key === 'Enter') && activeIndex < 0) {
+          return;
+        }
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        onSelectKeydown?.({ event: e.nativeEvent } as any);
+      }}
+      // @ts-ignore: Ignoring because the TS fails for React Native and works for web
+      onBlur={(e) => {
+        const nextItem = e.relatedTarget;
+        const nextItemRole = nextItem?.getAttribute('role');
+        if (nextItemRole !== 'combobox' && nextItemRole !== 'option') {
+          setIsOpen(false);
+        }
+      }}
+    >
+      {props.leading ? <Box>{props.leading}</Box> : null}
+      {props.title ? (
+        <Box paddingLeft="spacing.3" paddingRight="spacing.3">
+          <Text variant="caption" color="surface.text.subdued.lowContrast">
+            {props.title}
+          </Text>
+        </Box>
+      ) : null}
+      {props.trailing ? (
+        <Box
+          display="flex"
+          alignItems="center"
+          marginLeft={isOnlyActionButton ? undefined : 'auto'}
+          width={isOnlyActionButton ? '100%' : undefined}
+        >
+          {props.trailing}
+        </Box>
+      ) : null}
+    </StyledActionListFooter>
+  );
+};
+
+ActionListFooter.componentId = componentIds.ActionListFooter;
```

Comment:
Don't know how it was working lol. added WithComponentId utility anyway.

Reply:
> Don't know how it was working lol

TypeScript can infer/introspect static methods, though it's not recommended to do so, since once you generate the `.d.ts` files the typedefs won't have this.